src/value/ser.rs

```diff
@@ -9,8 +9,17 @@ impl Serialize for Value {
         match *self {
             Value::Null => serializer.serialize_unit(),
             Value::Bool(b) => serializer.serialize_bool(b),
-            Value::I64(i) => serializer.serialize_i64(i),
-            Value::F64(f) => serializer.serialize_f64(f),
+            Value::Number(ref n) => {
+                if let Some(u) = n.as_u64() {
+                    serializer.serialize_u64(u)
+                } else if let Some(i) = n.as_i64() {
+                    serializer.serialize_i64(i)
+                } else if let Some(f) = n.as_f64() {
+                    serializer.serialize_f64(f)
+                } else {
+                    panic!("unexpected number")
```

Comment:
I don't think this should panic. It's either `unreachable!()` or an error we should propagate.